client/state/add-reducer.js

```diff
@@ -3,24 +3,41 @@
 /**
  * Internal Dependencies
  */
-
 import { APPLY_STORED_STATE } from 'state/action-types';
 import { getStateFromLocalStorage } from 'state/initial-state';
 
+const initializations = {};
+
+function normalizeKey( key ) {
+	return Array.isArray( key ) ? key.join( '.' ) : key;
+}
+
+async function initializeState( store, reducer, storageKey ) {
+	const storedState = await getStateFromLocalStorage( reducer, storageKey );
+
+	if ( storedState ) {
+		store.dispatch( { type: APPLY_STORED_STATE, storageKey, storedState } );
+	}
+}
+
 // For a given store, creates a function that adds a new reducer to the store,
 // and loads (asynchronously) and applies the persisted state for it.
 export const addReducerToStore = store => async ( key, reducer ) => {
-	store.addReducer( key, reducer );
-
 	const { storageKey } = reducer;
-	if ( ! storageKey ) {
-		return;
-	}
+	const normalizedKey = normalizeKey( key );
+	let init = initializations[ normalizedKey ];
```

Comment:
Now when we can call `addReducerToStore` more than once, an additional check would be nice: verify that the reducer passed in is identical to the previous one.
```js
addReducerToStore( [ 'reader' ], readerReducer );
addReducerToStore( [ 'reader' ], anotherReducer );
```
will ignore the second call, but someone could reasonably expect to have the reducer replaced. Which we don't support. Better to thrown an exception.

Reply:
I could try that, yes. I'm somewhat concerned about the reducer identity, since while modules get deduplicated, I'm not sure if WebPack guarantees that they'll be the same object. I guess we can find out :)

Reply:
Done.